src/sql/workbench/contrib/notebook/find/notebookFindModel.ts

```diff
@@ -585,14 +584,59 @@ export class NotebookFindModel extends Disposable implements INotebookFindModel
 	// In markdown links are defined as [Link Text](https://url/of/the/text). when searching for text we shouldn't
 	// look for the values inside the (), below regex replaces that with just the Link Text.
 	cleanMarkdownLinks(cellSrc: string): string {
-		return cellSrc.replace(/(?:__|[*#])|\[(.*?)\]\(.*?\)/gm, '$1');
+		return cellSrc.replace(/(?:__)|\[([\s\S]*?)\]\(.*?\)/gm, '$1');
 	}
 
-	// remove /n's to calculate the line number to locate the correct element
+	// clean up the source w.r.t rendered html elements to locate the correct element
 	cleanUpCellSource(cellValue: string | string[]): string | string[] {
 		let trimmedCellSrc: string[] = [];
+		// regex to get the elements that span multiple lines into one.
+		const newline = new RegExp(/^([-=]+[\\r\\n]*)$/gm);
```

Comment:
Move these outside the class - recreating them every time the method is called is just wasteful